modules/lib/core/index.d.ts

```diff
@@ -52,23 +63,71 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+export interface Fragment {
+    type: 'fragment'
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
-    regions: Regions;
 }
 
-export interface Region<Config extends object = object> {
+export type LayoutDescriptor = keyof XpLayoutMap;
+export interface Layout<
+    Descriptor extends LayoutDescriptor = LayoutDescriptor
+> {
+    type: 'layout'
+    descriptor: Descriptor
+    config: XpLayoutMap[Descriptor]
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    regions: Record<string, Region<(Fragment | Part | TextComponent)[]>>;
+}
+
+export type PartDescriptor = keyof XpPartMap;
+export interface Part<
+    Descriptor extends PartDescriptor = PartDescriptor,
+> {
+    type: 'part'
+    descriptor: Descriptor
+    config: XpPartMap[Descriptor]
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+}
+
+export type PageDescriptor = keyof XpPageMap;
+export interface Page<
+    Descriptor extends PageDescriptor = PageDescriptor,
+> {
+    type: 'page'
+    descriptor: Descriptor
+    config: XpPageMap[Descriptor]
+    path: '/'
+    regions: Record<string, Region<(Fragment | Layout | Part | TextComponent)[]>>;
+}
```

Comment:
I think I would prefer to make these types completely independent of `XpPageMap` etc, and Instead create a new type which iterates over the `XpPageMap` and uses this one to to build up that union. (See `EveryPage` below.)

```suggestion
export interface Page<
    Descriptor extends string,
    Config
> {
    type: 'page'
    descriptor: Descriptor
    config: Config
    path: '/'
    regions: Record<string, Region<(Fragment | Layout | Part | TextComponent)[]>>;
}
```